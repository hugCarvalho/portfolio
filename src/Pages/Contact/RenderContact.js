import React from "react";
import { IsActiveContext } from "../../components/Header/NavButtons/RenderNavButtons";
import { server } from "../../config/server";
import "./RenderContact.scss";

function Contact() {
  const pageIsActive = React.useContext(IsActiveContext);

  React.useEffect(()=> {
    fetch(`${server}/contact`, {
      method: "POST",
      headers: {
        'Content-Type': 'application/json'
      },
  })
  }, [])

  React.useEffect(() => {
    document.title = "Contact";
  }, [pageIsActive]);

  return (
    <main className="Contact">
      <address className="contact-me">
        <a href="mailto:hugo.berlinlife@gmail.com">
          <i className="fas fa-envelope-square"></i>
        </a>
        <a href="mailto:hugo.berlinlife@gmail.com">hugo.berlinlife@gmail.com</a>
      </address>

      <address className="contact-me">
        <i className="fas fa-phone-square-alt"></i>
        <a href="tel:+4917678293187">+49 (0)176 782 93 187</a>
      </address>
    </main>
  );
}

export default Contact;
