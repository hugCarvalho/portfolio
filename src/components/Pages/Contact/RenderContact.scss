.Contact {
  background-color: hsla(150, 4%, 79%, 0.9);
  background-image: url("./assets/12.png");
  //background-image: url("../Skills/assets/10.png");

  min-height: calc(
    100vh - 234px
  ); //100vh - (header-height = 155 + footer-height = 70 extra=10)
  padding: 70px 16%;
  font-size: 2.5rem;

  display: flex;
  flex-direction: column;

  .contact-me {
    padding: 20px 10px 0 5%;

    display: flex;
    align-items: center;
  }

  a:hover {
    text-decoration: underline;
  }

  //Icons
  .fas,
  .fab {
    padding-right: 10px;
    font-size: 4rem;
  }
}

@media (max-width: 622px) {
  .Contact {
    // min-height: calc(100vh - 245px);
    padding: 40px 3%;
    font-size: 2rem;
  }
}
