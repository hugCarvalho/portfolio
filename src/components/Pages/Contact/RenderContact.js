import React from "react";
import "./RenderContact.scss";
// import { LanguageContext } from "../../../App";

function Contact() {
  // const { isLanguageEnglish } = React.useContext(LanguageContext);

  return (
    <div className="container__contact">
      {/* <h1>{isLanguageEnglish ? "Contact me" : "Kontakt mich"}</h1> */}
      <section className="wrapper__contact-info">
        <div className="contact-me">
          <a href="mailto:hugo.berlinlife@gmail.com">
            <i className="fas fa-envelope-square"></i>
          </a>
          <a href="mailto:hugo.berlinlife@gmail.com">
            hugo.berlinlife@gmail.com
          </a>
        </div>

        <div className="contact-me">
          <i className="fas fa-phone-square-alt"></i>
          <span>Telephone</span>
        </div>
      </section>
    </div>
  );
}

export default Contact;
