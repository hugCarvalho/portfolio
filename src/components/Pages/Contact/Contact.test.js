import React from "react";
import { render, screen } from "@testing-library/react";
import RenderContact from "./RenderContact";

test("should match snapshot", () => {
  const { container } = render(<RenderContact />);
  expect(container).toMatchSnapshot();
});

test("email link is correct", () => {
  render(<RenderContact />);
  const mailLink = screen.getByRole("link", { name: /hugo\.berlinlife@gmail\.com/i });
  expect(mailLink.href).toBe("mailto:hugo.berlinlife@gmail.com");
});
