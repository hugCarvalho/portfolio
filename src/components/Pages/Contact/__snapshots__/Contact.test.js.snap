// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <main
    class="Contact"
  >
    <address
      class="contact-me"
    >
      <a
        href="mailto:hugo.berlinlife@gmail.com"
      >
        <i
          class="fas fa-envelope-square"
        />
      </a>
      <a
        href="mailto:hugo.berlinlife@gmail.com"
      >
        hugo.berlinlife@gmail.com
      </a>
    </address>
    <address
      class="contact-me"
    >
      <i
        class="fas fa-phone-square-alt"
      />
      <a
        href="tel:+4917678293187"
      >
        +49 (0)176 782 93 187
      </a>
    </address>
  </main>
</div>
`;
